addons/project/models/project_task.py

```diff
@@ -1326,6 +1326,10 @@ def write(self, vals):
         elif 'project_id' in vals:
             self.filtered(lambda t: t.state != '04_waiting_normal').state = '01_in_progress'
 
+        # Do not recompute the state when changing the parent (to avoid resetting the state)
+        if 'parent_id' in vals:
```

Comment:
How does it work if we also change the project for instance? 🤔 

(I guess for the moment, the state will be recomputed before calling write method but I prefer asking now to be sure we will not introduced any regression)